src/Tests/Tests/Test_Vibrato.cpp

```diff
@@ -16,6 +15,314 @@ namespace vibrato_test {
+            iNumChannels = 1;
+            vibrato = nullptr;
+        }
+
+        ~VibratoTest() override {
+            // You can do clean-up work that doesn't throw exceptions here.
+        }
+
+        // If the constructor and destructor are not enough for setting up
+        // and cleaning up each test, you can define the following methods:
+
+        void SetUp() override {
+            // Code here will be called immediately after the constructor (right
+            // before each test).
+            CVibrato::create(vibrato, fDelayInSec, fWidthInSec, fSampleRateInHz, fFrequencyInHz, iNumChannels);
+            ppfInputBuffer = new float*[iNumChannels];
+            ppfOutputBuffer = new float*[iNumChannels];
+            for (int channel = 0; channel < iNumChannels; channel++)
+            {
+                ppfInputBuffer[channel] = new float[iBufferLength];
+                ppfOutputBuffer[channel] = new float[iBufferLength];
+            }
+        }
+
+        void TearDown() override {
+            // Code here will be called immediately after each test (right
+            // before the destructor).
+            CVibrato::destroy(vibrato);
+            for (int channel = 0; channel < iNumChannels; channel++)
+            {
+                delete[] ppfInputBuffer[channel];
+                delete[] ppfOutputBuffer[channel];
+            }
+            delete[] ppfInputBuffer;
+            delete[] ppfOutputBuffer;
+            ppfInputBuffer = nullptr;
+            ppfOutputBuffer = nullptr;
+        }
+
+        // Class members declared here can be used by all tests in the test suite
+        CVibrato *vibrato;
+        float fDelayInSec;
+        float fWidthInSec;
+        float fSampleRateInHz;
+        float fFrequencyInHz;
+        int iNumChannels;
+        float** ppfInputBuffer = nullptr;
+        float** ppfOutputBuffer = nullptr;
+        int iBufferLength = 1024;
+    };
+
+    TEST_F(VibratoTest, SetParamTest)
+    {
+        EXPECT_EQ(vibrato->setParam(CVibrato::kFrequency, -1.5f), Error_t::kFunctionInvalidArgsError);
+        EXPECT_EQ(vibrato->setParam(CVibrato::kDelay, -0.1f), Error_t::kFunctionInvalidArgsError);
+        EXPECT_EQ(vibrato->setParam(CVibrato::kWidth, -0.1f), Error_t::kFunctionInvalidArgsError);
+    }
+    TEST_F(VibratoTest, SetWidthGreaterThanDelay)
+    {
+        vibrato->setParam(CVibrato::kDelay, 0.1f);
+        EXPECT_EQ(vibrato->setParam(CVibrato::kWidth, 0.2f), Error_t::kFunctionInvalidArgsError);
+    }
+
+    TEST_F(VibratoTest, GetEqualsSet)
+    {
+        vibrato->setParam(CVibrato::kDelay, 0.1f);
+        EXPECT_NEAR(vibrato->getParam(CVibrato::kDelay), 0.1f, 1e-4);
+
+        vibrato->setParam(CVibrato::kWidth, 0.05f);
+        EXPECT_NEAR(vibrato->getParam(CVibrato::kWidth), 0.05f, 1e-4);
+
+        vibrato->setParam(CVibrato::kFrequency, 42.f);
+        EXPECT_NEAR(vibrato->getParam(CVibrato::kFrequency), 42.f, 1e-4);
+    }
+
+    TEST_F(VibratoTest, BlockSizeTest)
```

Comment:
This test required you to have varying input block sizes, you have only given it a random process block size, which does not cover the test case. Try to create a function called process in the VibratoTest class and use an array of blocksizes to create multiple input buffers for this test.